tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/SanitizerTests.cs

```diff
@@ -303,6 +303,19 @@ public void BodyKeySanitizerKeyReplace()
             Assert.Contains(replacementValue, Encoding.UTF8.GetString(targetEntry.Request.Body));
         }
 
+        [Fact]
+        public void BodyKeySanitizerHandlesNonJSON()
+        {
+            var session = TestHelpers.LoadRecordSession("Test.RecordEntries/oauth_request.json");
```

Comment:
I assume this session doesn't have a JSON body?

Reply:
That's exactly correct. Just deliberately making sure to exercise the new functionality 👍 

FYI the content of the body is: 

```
scope=https%3A%2F%2Fvault.azure.net%2F.default%20openid%20profile%20offline_access&client_id=04b07795-8ddb-461a-bbee-02f9e1bf7b46&grant_type=device_code&device_code=DEVICE_CODE&client-request-id=client-request-id&client_info=1&x-client-SKU=msal.js.node&x-client-VER=1.0.3&x-client-OS=linux&x-client-CPU=x64&x-ms-lib-capability=retry-after, h429&x-client-current-telemetry=2|671,0|,&x-client-last-telemetry=2|0|||0,0
```

So it actually is hitting the `catch` in debug.

Reply:
I also did validate that it is indeed a `JsonReaderException` from both `JsonConvert.DeserializeObject` and `JToken.Parse`.

Reply:
Yeah, that was the part I was wondering about since we actually don't check the exception in the Test Framework, but I think what you have is better 😄